### ISSUE TITLE
Java suppressing one rule suppresses the entire category

### ISSUE THREAD

**Issue by alanbuttars** [pmd/pmd#3102]
**Affects PMD Version:** 6.31.0

**Description:**

***1.*** Default settings
Using the below gradle config:

```gradle
pmd {
    consoleOutput = true
    toolVersion = '6.31.0'
}
```
I see some expected rule violations:
```
/path/to/File.java:241: AvoidDuplicateLiterals: The String literal "BLAH," appears 4 times in this file; the first occurrence is on line 241
/path/to/File.java:394: DataflowAnomalyAnalysis:        Found 'DU'-anomaly for variable 'myTime' (lines '394'-'420').
/path/to/File.java:26:  BeanMembersShouldSerialize:     Found non-transient, non-static member. Please mark as transient or provide accessors.
```

***2.*** Custom settings
I want to suppress the `BeanMembersShouldSerialize` violations, so I try:

```gradle
pmd {
    consoleOutput = true
    ruleSets = [] // Clear the default rules
    ruleSetConfig = resources.text.fromFile('config/pmd/rule-set.xml') // Override with custom rules
    toolVersion = '6.31.0'
}
```

My `rule-set.xml` is:
```xml
<?xml version="1.0"?>
<ruleset
    xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
    <rule ref="category/java/errorprone.xml">
        <exclude name="BeanMembersShouldSerialize" />
    </rule>
</ruleset>
```

This successfully suppresses `BeanMembersShouldSerialize`, but it also suppresses `DataflowAnomolyAnalysis`. I think this is because both of those rules live in `category/java/errorprone.xml`:

```
/path/to/File.java:241: AvoidDuplicateLiterals: The String literal "BLAH," appears 4 times in this file; the first occurrence is on line 241
```

**Running PMD through:** *[Gradle]*

**Comment by jsotuyod** [comment]
@alanbuttars which version of Gradle are you using?

`DataflowAnomalyAnalysis` is ignored by default because the rule is deprecated (check the second notice [here](https://pmd.github.io/pmd-6.31.0/pmd_userdocs_making_rulesets.html#bulk-adding-rules)). You would have to explicitly include it for it to be executed.

You may be seeing it by default if you are using an older Gradle version, which would default to referencing the old (deprecated) rulesets, instead of rules by categories.

**Comment by alanbuttars** [comment]
Aha! That makes sense. I'm using Gradle 5.1.
